test/UnitTests/MSTest.Analyzers.UnitTests/TestMethodShouldNotBeIgnoredAnalyzerTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Framework;
+using Microsoft.Testing.TestInfrastructure;
+
+using VerifyCS = MSTest.Analyzers.Test.CSharpCodeFixVerifier<
+    MSTest.Analyzers.TestMethodShouldNotBeIgnoredAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace MSTest.Analyzers.Test;
+
+[TestGroup]
+public sealed class TestMethodShouldNotBeIgnoredAnalyzerTests(ITestExecutionContext testExecutionContext) : TestBase(testExecutionContext)
+{
+    public async Task WhenTestMethodIsNotIgnored_NoDiagnostic()
+    {
+        var code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [TestMethod]
+                public void MyTestMethod()
+                {
+                }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(code);
+    }
+
+    public async Task UsingIgnoreWithoutTestMethod_NoDiagnostic()
+    {
+        var code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [Ignore]
```

Comment:
I don't recall, have we implemented a rule that raises in this case? If not probably worth adding the point to the big ticket with list of rules.

Reply:
when exactly we have to raise? test class without a test method?

Reply:
I think amaury means this rule, that checks that additional test attributes  (Like TestCategory),are only used on methods that have [TestMethod]. And I don't see Ignore in that list of attributes that are checked.

https://learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0007